controller/registries/consul/config.go

```diff
@@ -181,46 +258,145 @@ func (reg *Consul) fetchAllServices(client *Client) (map[consulService]bool, err
 	}
 	serviceMap := make(map[consulService]bool)
 	for serviceName, tags := range services {
-		for _, tag := range tags {
-			service := consulService{
-				Tag:         tag,
-				ServiceName: serviceName,
-			}
-			serviceMap[service] = true
+		tag := strings.Join(tags, "-")
+		service := consulService{
+			Tag:         tag,
+			ServiceName: serviceName,
 		}
+		serviceMap[service] = true
 	}
+
 	return serviceMap, nil
 }
 
-func (reg *Consul) subscribe(serviceName string) error {
-	fmt.Println(serviceName)
+func (reg *Consul) getServiceEntryKey(tag, serviceName string) string {
+	var host string
+	suffix := strings.Join([]string{reg.client.NameSpace, reg.client.DataCenter, reg.name, RegistryType}, ".")
+	suffix = strings.ReplaceAll(suffix, "_", "-")
+	if len(tag) > 0 {
```

Comment:
Join an empty string is harmless as we will remove the repeated dots below. We can remove an extra branch.